src/site/content/en/blog/device-pixel-content-box/index.md

```diff
@@ -0,0 +1,135 @@
+
+<figure class="w-figure">
+  <img src="side-by-side.png" alt="" class="w-screenshot" width="600">
+  <figcaption class="w-figcaption">The top image is a raster of differently colored pixels. The bottom image is the same as above, but the width and height have been reduced by one pixel using bilinear scaling. The emerging pattern is called the Moiré effect.<br>(You might have to open this image in a new tab to see it without any scaling applied to it.)</figcaption>
+</figure>
+
+## devicePixelContentBox
+
+At this point, `devicePixelContentBox` should make a lot more sense to you. It gives you an element's content box in device pixel (i.e. physical pixel) units. It's part of `ResizeObserver`. While [ResizeObserver is now supported in all major browsers][ro support] since Safari 13.1, the `devicePixelContentBox` property is only in Chrome 84+ for now.
+
+As mentioned in [the main `ResizeObserver` article][resizeobserver], the `ResizeObserver`'s callback function will be called before paint and after layout. That means that the `entries` parameter to the callback will contain the sizes of all observed elements just before they are being painted. In the context of our canvas problem outlined above, we can use this opportunity to adjust the number of pixels on our canvas, ensuring that we end up with an exact one-to-one mapping between canvas pixels and physical pixels.
+
+```js
+const observer = new ResizeObserver((entries) => {
+  const entry = entries.find((entry) => entry.target === canvas);
+  canvas.width = entry.devicePixelContentBoxSize[0].inlineSize;
+  canvas.height = entry.devicePixelContentBoxSize[0].blockSize;
+
+  /* … render to canvas … */
+});
+observer.observe(canvas, {box: ['device-pixel-content-box']});
+```
+
+The `box` property in the options object for `observer.observe()` lets you define which sizes you wish to _observe_. So while each `ResizeObserverEntry` will always provide `borderBoxSize`, `contentBoxSize` and `devicePixelContentBoxSize` (provided the browser supports it), the callback will only be invoked if any of the _observed_ box metrics change.
+
+**Note:** All of the box metrics are arrays to allow `ResizeObserver` to handle fragmentation in the future. At the time of writing, the array is always of length 1.
+
+With this new property, we can even animate our canvas' size and position (effectively guaranteeing fractional pixel values), and not see any Moiré effects on the rendering. If you would like to see the Moiré effect on the approach using `getBoundingClientRect()`, and how the new `ResizeObserver` property allows you to avoid, take a look at the [demo] in Chrome Canary!
```

Comment:
```suggestion
With this new property, we can even animate our canvas' size and position (effectively guaranteeing fractional pixel values), and not see any Moiré effects on the rendering. If you would like to see the Moiré effect on the approach using `getBoundingClientRect()`, and how the new `ResizeObserver` property allows you to avoid it, take a look at the [demo] in Chrome 84 or later!
```